KIPs/kip-xxx.md

```diff
@@ -0,0 +1,178 @@
+
+Note that whether the emptiness of storageRoot does not matter in transaction type restrictions.
+
+### Changes to APIs
+
+#### eth_getRawTransaction, eth_sendRawTransaction
+
+eth_getRawTransaction returns the raw transaction RLP prefixed with `SET_CODE_TX_TYPE (0x04)` and eth_sendRawTransaction accepts such raw transaction.
+
+#### kaia_getRawTransaction, kaia_sendRawTransaction
+
+kaia_getRawTransaction returns the raw transaction RLP prefixed with `ENVELOPED_SET_CODE_TX_TYPE (0x7804)` and kaia_sendRawTransaction accepts such raw transaction.
+
+#### eth_getBlock*, eth_getTransaction*, eth_getTransactionReceipt
+
+For SetCode transaction type, eth namespace APIs that return the transaction fields MUST show the `authorizationList` field and its `type: "0x04"`.
+
+#### kaia_getBlock*, kaia_getTransaction*, kaia_getTransactionReceipt
+
+For SetCode transaction type, kaia namespace APIs that return the transaction fields MUST show the `authorizationList` field and its `typeInt: 30724`, and `type: "TxTypeEthereumSetCode"`.
+
+#### eth_getCode, eth_getStorageAt, kaia_getCode, kaia_getStorageAt
+
+APIs querying the code and storage of an account MUST work correctly for EOAs.
+
+#### kaia_getAccount
+
+For both EOAs and SCAs, its kaia_getAccount result SHOULD include storageRoot, codeHash, codeFormat, vmVersion fields.
+
+### Changes to SDK
+
+Before an SDK signs an authorization tuple, the SDK MAY verify that the authority’s AccountKey is of type AccountKeyTypeLegacy. If the SDK does not verify, it is the responsibility of the SDK user (e.g. Wallet) to make sure the authority is eligible to use the SetCode transaction.
+
+## Rationale
+
+### Extending the EOA instead of converting to SCA
+
+Because the feature is called “Set EOA account code”, there could be a need to distinguish between EOA with code and an SCA.
+
+### EOAs with updated AccountKeys cannot sign authorization
+
+In Kaia, EOAs may change the transaction signing permissions by sending an AccountUpdate transaction. But in this KIP, we strictly let the original private key (i.e. the one that AccountKeyLegacy accepts) to sign the SetCode authorization. Decoupling the two permissions will be confusing, so EOAs with code must have AccountKeyLegacy at all times.
+
+### EOA RLP formats depend on field values
+
+For backward compatibility, we have to retain the compact encoding.
+
+If we choose to select the encoding depending on hardfork instead, then all trie handling code must be fork-aware, and coding in that is non-trivial.
+
+### Kaia transaction type restrictions
+
+ValueTransfer transaction types cannot be sent to EOAs with code. Kaia’s ValueTransfer transaction types have the semantics of simply increasing the balance of the destination EOA. If ValueTransfer can target EOA with code, the balance increase would bypass the EOA’s fallback() or receive() function. Opening a way to bypass the fallback may have an unknown security implication, thus prohibiting such a case.
```

Comment:
Are these Ethereum Types allowed?
- TxTypeLegacyTransaction
- TxTypeEthereumAccessList
- TxTypeEthereumDynamicFee

Reply:
The semantics of Eth txtypes shouldn't change. But since we do modify the vm.Call condition, let me clarify that in e.g. Backwards Compatibility section